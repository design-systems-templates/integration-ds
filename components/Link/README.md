# @integration-ds/link

Welcome to my `Link` component.

## Installation

```sh
npm i @integration-ds/link
# or with yarn
yarn add @integration-ds/link
```

Then to use the component in your code just import it!

```js
import Link from '@integration-ds/link';
// and with css-modules
import '@integration-ds/link/dist/main.css';
```
