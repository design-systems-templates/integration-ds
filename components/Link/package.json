{
  "name": "@integration-ds/link",
  "version": "0.0.0",
  "main": "./dist/cjs/index.js",
  "module": "./dist/esm/index.js",
  "types": "./dist",
  "repository": "https://github.com/design-systems-templates/integration-ds",
  "author": "Raj Vasikarla raj_vasikarla@intuit.com",
  "contributors": ["Raj Vasikarla raj_vasikarla@intuit.com"],
  "license": "MIT",
  "sideEffects": ["*.css"],
  "scripts": {
    "clean": "ds clean",
    "build": "ds build",
    "dev": "ds dev",
    "storybook": "ds storybook start",
    "start": "ds build --watch",
    "test": "ds test",
    "lint": "ds lint",
    "size": "ds size"
  },
  "dependencies": {
    "@babel/runtime": "7.7.2"
  },
  "peerDependencies": {
    "react": ">= 16.8.6"
  },
  "files": [
    "dist",
    "src",
    "!*.snippet.*",
    "!*theme.*",
    "!*.test.*",
    "!__snapshots__",
    "!__tests__"
  ]
}
