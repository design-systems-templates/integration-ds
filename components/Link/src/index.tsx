import React from 'react';
import styles from './Link.css';

interface LinkProps {
  /** The value for the component to display */
  value?: string;
}

/** A simple component. */
export const Link = (props: LinkProps) => {
  return <div className={styles.link}>This is a Link component {props.value}</div>;
}

export default Link;
