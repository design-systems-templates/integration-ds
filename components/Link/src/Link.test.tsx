import React from 'react';
import { render } from '@testing-library/react';
import Link from '.';

describe('Link', () => {
  test('It matches the snapshot', () => {
    const { container } = render(<Link />);
    expect(container).toMatchSnapshot();
  });  
});