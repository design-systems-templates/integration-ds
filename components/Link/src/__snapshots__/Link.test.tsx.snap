// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link It matches the snapshot 1`] = `
<div>
  <a
    class="link"
    href="http://www.google.com"
  >
    This is a Link component 
  </a>
</div>
`;
